spec/jobs/ra/fetch_changes_batch_job_spec.rb

```diff
@@ -0,0 +1,201 @@
+        and_return(fixture_filepath('jobs/ra/fixtures/DavkaInit_20151009-01.xml'))
+
+      subject.perform('abc', downloader: downloader)
+
+      region_change = Ra::RegionChange.first
+      expect(region_change).to have_attributes(
+        region_id: 1,
+        database_operation: 'INSERT',
+        version_id: 1,
+        created_reason: 'CREATE',
+        valid_to: Time.parse('2004-04-30T23:59:59+02:00'),
+        effective_on: Date.parse('1996-07-24+02:00'),
+      )
+
+      expect(region_change.region_code).to have_attributes(
+        code: '0',
+        name: 'Neznámy'
+      )
+    end
+
+    it 'can parse county changes' do
+      expect(downloader).to receive(:download_file).with('abc').
+        and_return(fixture_filepath('jobs/ra/fixtures/DavkaInit_20151009-02.xml'))
+
+      subject.perform('abc', downloader: downloader)
+
+      county_change = Ra::CountyChange.first
+      expect(county_change).to have_attributes(
+        county_id: 11,
+        region_id: 1,
+        database_operation: 'INSERT',
+        version_id: 1,
+        created_reason: 'CREATE',
+        valid_to: Time.parse('2004-04-30T23:59:59+02:00'),
+        effective_on: Date.parse('1996-07-24+02:00'),
+      )
+
+      expect(county_change.county_code).to have_attributes(
+        code: '6000',
+        name: 'Neznámy'
+      )
+    end
+
+    it 'can parse municipality changes' do
+      expect(downloader).to receive(:download_file).with('abc').
+        and_return(fixture_filepath('jobs/ra/fixtures/DavkaInit_20151009-03.xml'))
+
+      subject.perform('abc', downloader: downloader)
+
+      municipality_change = Ra::MunicipalityChange.first
+      expect(municipality_change).to have_attributes(
+        municipality_id: 1,
+        county_id: 11,
+        database_operation: 'INSERT',
+        version_id: 1,
+        created_reason: 'CREATE',
+        valid_to: Time.parse('2004-04-30T23:59:59+02:00'),
+        effective_on: Date.parse('1996-07-24+02:00'),
+        municipality_status: 'MUNICIPALITY'
+      )
+
+      expect(municipality_change.municipality_code).to have_attributes(
+        name: 'Neznáma'
+      )
+    end
+
+    it 'can parse district changes' do
+      expect(downloader).to receive(:download_file).with('abc').
+        and_return(fixture_filepath('jobs/ra/fixtures/DavkaInit_20151009-04.xml'))
+
+      subject.perform('abc', downloader: downloader)
+
+      district_change = Ra::DistrictChange.first
+      expect(district_change).to have_attributes(
+        district_id: 4764,
+        municipality_id: 92,
+        database_operation: 'INSERT',
+        version_id: 1,
+        created_reason: 'CREATE',
+        valid_to: Time.parse('2004-04-30T23:59:59+02:00'),
+        unique_numbering: false
+      )
+
+      expect(district_change.district_code).to have_attributes(
+        code: '219755',
+        name: 'Nivy'
+      )
+    end
+
+    pending 'it adds batch record last'
```

Comment:
@eronisko - toto imho len cele zabalit do transakcie. bude to long running tx ale nic sa neda velmi robit.